src/components/PokeCard.js

```diff
@@ -0,0 +1,189 @@
+  }
+
+  function stylesNormalCard() {
+    return {
+      transform: "rotateY(0deg) rotateX(0deg)",
+      transition: "transform 0.5s ease, box-shadow 0.5s ease",
+      boxShadow: "0 10px 5px rgba(0, 0, 0, 0.2)"
+    };
+  }
+
+  function styleNormalSprite() {
+    return {
+      transform: "translateZ(0px) rotateZ(0deg)"
+    };
+  }
+
+  function styleNormalText() {
+    return { transform: "translateZ(0px)" };
+  }
+
+  function stylePerspectiveCard(xAxis, yAxis) {
+    return {
+      transform: `rotateY(${xAxis}deg) rotateX(${yAxis}deg) scale(1.1)`,
+      transition: "box-shadow 0.5s ease",
+      boxShadow: "0 20px 20px rgba(0, 0, 0, 0.2)"
+    };
+  }
+
+  function stylePerspectiveSprite(xAxis, yAxis) {
+    return {
+      transform: "translateZ(130px) rotateZ(-1deg) scale(1.09)",
+      filter: `drop-shadow(${-xAxis * 1.5}px ${
+        yAxis * 1.5
+      }px 5px rgba(0, 0, 0, 0.6))`
+    };
+  }
+
+  function stylePerspectiveText(xAxis, yAxis) {
+    return {
+      transform: "translateZ(40px)",
+      filter: `drop-shadow(${-xAxis / 2}px ${
+        yAxis / 2
+      }px 1px rgba(0, 0, 0, 0.1))`
+    };
+  }
+
+  return (
+    <div style={styles.container}>
+      <Card
+        parentRef={cardRef}
+        dynamicStyle={cardStyle}
+        onMouseMove={onMoveCard}
+        onMouseLeave={onLeaveCard}
+      >
+        <CardTitle dynamicStyle={textStyle}>{mockPokemonData.name}</CardTitle>
+        <CardSprites
+          dynamicStyle={spriteStyle}
+          sprites={[
+            {
+              img: mockPokemonData.sprites.front_default,
+              alt: mockPokemonData.name
+            },
+            {
+              img: mockPokemonData.sprites.front_shiny,
+              alt: mockPokemonData.name
+            }
+          ]}
+        />
+        <CardLinkVideo href={mockPokemonData.video} dynamicStyle={textStyle}>
+          {mockPokemonData.name} video
+        </CardLinkVideo>
+      </Card>
+    </div>
+  );
+}
+
+const Card = (props) => {
+  return (
+    <div
+      ref={props.parentRef}
+      style={{ ...styles.card.container, ...props.dynamicStyle }}
+      onMouseMove={props.onMouseMove}
+      onMouseLeave={props.onMouseLeave}
+    >
+      {props.children}
+    </div>
+  );
+};
+
+const CardTitle = (props) => {
+  return (
+    <h1 style={{ ...styles.card.text, ...props.dynamicStyle }}>
+      {props.children}
+    </h1>
+  );
+};
+
+const CardSprites = (props) => {
+  return (
+    <div>
+      {props.sprites.map((sprite, i) => {
+        return (
+          <img
+            key={i}
+            src={sprite.img}
+            alt={props.alt}
+            style={{ ...styles.card.img, ...props.dynamicStyle }}
+          />
+        );
+      })}
+    </div>
+  );
+};
+
+const CardLinkVideo = (props) => {
+  return (
+    <p style={{ ...styles.card.text, ...props.dynamicStyle }}>
+      <a
+        href={props.href}
+        target="_blank"
+        rel="noreferrer"
+        style={styles.card.text}
```

Comment:
It is better to style components using classnames instead

Reply:
It's done, now I looks better :D